packages/db/config/configValue.ts

```diff
@@ -0,0 +1,66 @@
+import { z } from "zod";
+
+export enum ConfigType {
+  BOOLEAN,
+  STRING,
+  NUMBER,
+  INFERENCE_PROVIDER_ENUM,
+}
+
+export enum InferenceProviderEnum {
+  DISABLED = "disabled",
+  OPEN_AI = "openai",
+  OLLAMA = "ollama",
+}
+
+export const InferenceProviderEnumValidator = z.enum([
+  InferenceProviderEnum.DISABLED,
+  InferenceProviderEnum.OPEN_AI,
+  InferenceProviderEnum.OLLAMA,
+]);
+
+export type InferenceProviderEnumZodType =
+  typeof InferenceProviderEnumValidator;
+
+export interface ConfigTypeMap {
+  [ConfigType.BOOLEAN]: boolean;
+  [ConfigType.STRING]: string;
+  [ConfigType.NUMBER]: number;
+  [ConfigType.INFERENCE_PROVIDER_ENUM]: string;
+}
+
+export interface ConfigValidatorMap {
+  [ConfigType.BOOLEAN]: z.ZodBoolean;
+  [ConfigType.STRING]: z.ZodString;
+  [ConfigType.NUMBER]: z.ZodNumber;
+  [ConfigType.INFERENCE_PROVIDER_ENUM]: InferenceProviderEnumZodType;
+}
+
+export interface ConfigProperties<T extends ConfigType> {
+  key: string;
+  name: string;
+  type: T;
+  defaultValue: ConfigTypeMap[T];
+  validator: ConfigValidatorMap[T];
+}
+
+export class ConfigValue<T extends ConfigType> {

```

Comment:
Why not `ConfigValue extends ConfigProperties<T extends ConfigType>` to remove the duplicate properties?